site/en/blog/immutable-document-domain/index.md

```diff
@@ -0,0 +1,197 @@
+  Beginning with Chrome version 101, websites will be unable to set `document.domain`.
+date: 2022-01-06
+authors:
+  - agektmr
+tags:
+  - security
+hero: 'image/YLflGBAPWecgtKJLqCJHSzHqe2J2/grGMzuhOjsGhN150dONe.jpg'
+alt: >
+  A dog in disguise.
+---
+
+**If your website relies on relaxing the same-origin policy via `document.domain`,
+your action is required.**
+
+Beginning with Chrome version 101, websites will be unable to set
+`document.domain`. Websites will need to use alternative approaches such as
+`postMessage()` or Channel Messaging API to communicate cross-origin. If a
+website relies on same-origin policy relaxation via `document.domain` to
+function correctly, it will need to send an `Origin-Agent-Cluster: ?0` header
+along with all documents that require that behavior (note that `document.domain`
+has no effect if only one document sets it).
+
+## Why make `document.domain` immutable?
+
+Many websites set `document.domain` to allow communication between [same-site
+but cross-origin](https://web.dev/same-site-same-origin/) pages. 
+
+{% Aside 'key-term' %}
+
+Sites which have the same
+[eTLD+1](https://web.dev/same-site-same-origin/#:~:text=the%20whole%20site%20name%20is%20known%20as%20the%20etld%2B1)
+but are different subdomains are considered "same-site but cross-origin".
+
+{% endAside %}
+
+Here's how it's used:
+
+Let's say a page on `https://parent.example.com` embeds an iframe page from
+`https://video.example.com`. These pages have the same eTLD+1 (`example.com`)
+with different subdomains. By setting both pages' `document.domain` to
+`'example.com'`, the two origins can be treated as if they are the same-origin
+by the browser.
+
+On `https://parent.example.com`:
+
+```js
+console.log(document.domain); // "parent.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://video.example.com`:
+
+```js
+console.log(document.domain); // "video.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://parent.example.com`, do a cross origin DOM manipulation against
+`https://video.example.com`.
+
+Websites set `document.domain` to make it possible for same-site documents to
+communicate more easily. Because this change [relaxes the same-origin
+policy](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction),
+the parent page will be able to access the iframe's document and traverse the
+DOM tree, and vice versa.
+
+At a glance, this is a convenient technique, but this actually introduces a
+security risk.
+
+### Security concerns with `document.domain`
+
+Security concerns have led to the spec for `document.domain` to [warn users to
+avoid
+usage](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction).
+The [current discussion with other browser
+vendors](https://github.com/w3ctag/design-reviews/issues/564) is moving toward
+the same direction.
```

Comment:
```suggestion
Security concerns around `document.domain` have led to a change in the [specification that warns users to
avoid
using it](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction).
The [current discussion with other browser
vendors](https://github.com/w3ctag/design-reviews/issues/564) is moving in
the same direction.
```

Reply:
Editing tip: "spec" is basically slang, so always use "specification" in public communications.